syncstorage/tests/functional/test_storage.py

```diff
@@ -1636,21 +1656,69 @@ def test_batch_with_immediate_commit(self):
           {'id': 'b', 'payload': 'bie'},
           {'id': 'c', 'payload': 'cee'}
         ]
-        resp = self.app.post_json(collection, bsos)
```

Comment:
This was a totally pointless additional post, which happened to trigger the bug!  I've moved it into a separate testcase for clarity.